shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/ShivaVariantcalling.scala

```diff
@@ -112,21 +112,22 @@ class ShivaVariantcalling(val root: Configurable) extends QScript
         vtDecompose.inputVcf = vtNormalize.outputVcf
         vtDecompose.outputVcf = swapExt(caller.outputDir, vtNormalize.outputVcf, ".vcf.gz", ".decompose.vcf.gz")
         add(vtDecompose, Tabix(this, vtDecompose.outputVcf))
-        cv.variant :+= TaggedFile(vtDecompose.outputVcf, caller.name)
+        if (caller.mergeVcfResults) cv.variant :+= TaggedFile(vtDecompose.outputVcf, caller.name)
       } else if (normalize && !decompose) {
         vtNormalize.outputVcf = swapExt(caller.outputDir, caller.outputFile, ".vcf.gz", ".normalized.vcf.gz")
         add(vtNormalize, Tabix(this, vtNormalize.outputVcf))
-        cv.variant :+= TaggedFile(vtNormalize.outputVcf, caller.name)
+        if (caller.mergeVcfResults) cv.variant :+= TaggedFile(vtNormalize.outputVcf, caller.name)
       } else if (!normalize && decompose) {
         vtDecompose.inputVcf = caller.outputFile
         vtDecompose.outputVcf = swapExt(caller.outputDir, caller.outputFile, ".vcf.gz", ".decompose.vcf.gz")
         add(vtDecompose, Tabix(this, vtDecompose.outputVcf))
-        cv.variant :+= TaggedFile(vtDecompose.outputVcf, caller.name)
-      } else cv.variant :+= TaggedFile(caller.outputFile, caller.name)
+        if (caller.mergeVcfResults) cv.variant :+= TaggedFile(vtDecompose.outputVcf, caller.name)
+      } else if (caller.mergeVcfResults) cv.variant :+= TaggedFile(caller.outputFile, caller.name)
+    }
+    if (cv.variant.nonEmpty) {
+      add(cv)
+      addStats(finalFile, "final")
```

Comment:
I think we should also collect stats for the cases when callsets do not get merged. In that case, there are two possibilities:

* There is only one callset, because only one caller was used, and the final file is simply the only VCF file. 
* There are multiple callsets, and stats should preferably be collected for all of them.